### ISSUE TITLE
Please add JPMS/module-info data to this project

### ISSUE THREAD

**Issue by Siedlerchr** [deepjavalibrary/djl#3473]
## Description

For better usage in a modularized application and to prevent hacky workarounds, it would be great to provide the correct module info declaration. Especially for the ServiceProviders.
Refs https://github.com/JabRef/jabref/pull/11761

https://www.oracle.com/corporate/features/understanding-java-9-modules.html

Will this change the current api? How?

No

Who will benefit from this enhancement?

Consumers of the libary e.g. JabRef
https://github.com/JabRef/jabref/pull/11761 

## References

Langchain: https://github.com/langchain4j/langchain4j/issues/1066
https://www.oracle.com/corporate/features/understanding-java-9-modules.html 
https://javanexus.com/blog/solving-puzzle-java-9-module-system-challenges
https://www.baeldung.com/java-modularity

**Comment by frankfliu** [comment]
@Siedlerchr 

We already support auto-module in DJL. Here is example: https://github.com/deepjavalibrary/djl-demo/tree/master/development/module

**Comment by Siedlerchr** [comment]
Yes, that's a good thing but explicit module would be better, especially regarding the zoo service provider
